pkg/local_workflows/data_transformation_workflow.go

```diff
@@ -108,63 +108,353 @@ func dataTransformationEntryPoint(invocationCtx workflow.InvocationContext, inpu
+			finding.Attributes.Policy = &local_models.TypesPolicyv1{
+				OriginalLevel:    &res.Properties.Policy.OriginalLevel,
+				OriginalSeverity: &res.Properties.Policy.OriginalSeverity,
+				Severity:         &res.Properties.Policy.Severity,
+			}
+		}
+
+		finding.Attributes.Rating = &local_models.TypesFindingRating{
+			Severity: struct {
+				OriginalValue *local_models.TypesFindingRatingSeverityOriginalValue `json:"original_value,omitempty"`
+				Reason        *local_models.TypesFindingRatingSeverityReason        `json:"reason,omitempty"`
+				Value         local_models.TypesFindingRatingSeverityValue          `json:"value"`
+			}{
+				Value: local_models.TypesFindingRatingSeverityValue(sarif_utils.SarifLevelToSeverity(res.Level)),
+			},
+		}
+
+		finding.Attributes.Rating.Priority = &local_models.TypesFindingNumericalRating{
+			Score: res.Properties.PriorityScore,
+			Factors: func() (factors []local_models.RiskFactors) {
+				for _, v := range res.Properties.PriorityScoreFactors {
+					factor := &local_models.RiskFactors{}
+					err := factor.FromTypesVulnerabilityFactRiskFactor(local_models.TypesVulnerabilityFactRiskFactor{
+						Name:  v.Type,
+						Value: v.Label,
+					})
+					if err != nil {
+						return nil
+					}
+					factors = append(factors, *factor)
+				}
+				return factors
+			}(),
+		}
+
+		codeFlows := mapCodeFlows(res)
+		finding.Attributes.CodeFlows = &codeFlows
+
+		finding.Attributes.Suggestions = &[]local_models.Suggestion{}
+		finding.Attributes.Locations = &[]local_models.IoSnykReactiveFindingLocation{}
+		*finding.Attributes.Locations = append(*finding.Attributes.Locations, mapLocations(res)...)
+		finding.Attributes.Suppression = mapSuppressions(res)
+
+		findings = append(findings, finding)
 	}
+	return findings, nil
+}
 
-	apiOutput, applyError := sarif2apiTransformer.Apply(input)
-	if applyError != nil {
-		return localFinding, applyError
+func mapSuppressions(res sarif.Result) *local_models.TypesSuppression {
+	if len(res.Suppressions) == 0 {
+		return nil
 	}
+	suppression := res.Suppressions[0]
+	expiration := ""
+	ignored_email := ""
+	if suppression.Properties.Expiration != nil {
+		expiration = *suppression.Properties.Expiration
+	}
+	if suppression.Properties.IgnoredBy.Email != nil {
+		ignored_email = *suppression.Properties.IgnoredBy.Email
+	}
+	return &local_models.TypesSuppression{
+		Details: &local_models.TypesSuppressionDetails{
+			Category:   string(suppression.Properties.Category),
+			Expiration: expiration,
+			IgnoredOn:  suppression.Properties.IgnoredOn,
+			IgnoredBy: local_models.TypesUser{
+				Name:  suppression.Properties.IgnoredBy.Name,
+				Email: ignored_email,
+			},
+		},
+		Justification: &suppression.Justification,
+		Kind:          "ignored",
+	}
+}
 
-	cliOutput, applyError := api2cliTransformer.ApplyValue(apiOutput)
-	if applyError != nil {
-		return localFinding, applyError
+func mapFingerprints(res sarif.Result) []local_models.Fingerprint {
```

Comment:
Suggestion: There is quite some repetition in this method, I would recommend to have the logic for converting logic only once and iterate over a list of fingerprints. This will also make future extensions easier.

Reply:
Suggestion: similar to a previous comment, I would recommend to use sarif fingerprints as an input and local findings fingerprints as an output.